test/infrastructure/docker/api/v1alpha4/hollowmachine_types.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha4
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// HollowMachineFinalizer allows Reconcileto clean up resources associated with HollowMachine before
+	// removing it from the apiserver.
+	HollowMachineFinalizer = "hollowmachine.infrastructure.cluster.x-k8s.io"
+)
+
+// HollowMachineSpec defines the desired state of HollowMachine.
+type HollowMachineSpec struct {
+	// ExternalCluster define the name of the cluster where Pods hosting hollow Kubelet should be executed.
+	// If empty, the machine's cluster will be used.
+	ExternalCluster *string `json:"externalCluster"`
+
+	// Pod allows to tune some details of the Pods hosting hollow Kubelet.
+	// +optional
+	Pod HollowPodSpec `json:"pod,omitempty"`
+
+	// ProviderID reports the value assigned by kubemark on the hollow Kubelet.
+	// +optional
+	ProviderID *string `json:"providerID,omitempty"`
+}
+
+// HollowPodSpec allows to tune some details of the Pods hosting hollow Kubelet.
+type HollowPodSpec struct {
+	// Containers allows to tune some details of the containers.
+	// +optional
+	Containers HollowContainersSpec `json:"containers,omitempty"`
+
+	// If specified, the pod's tolerations.
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+}
+
+// HollowContainersSpec allows to tune some details of the containers in the Pod hosting hollow Kubelet.
+type HollowContainersSpec struct {
+	// Kubemark allows to tune some details of the kubemark container.
+	// +optional
+	Kubemark KubemarkContainersSpec `json:"kubemark,omitempty"`
+}
+
+// KubemarkContainersSpec allows to tune some details of the kubemark container in the Pod hosting hollow Kubelet.
+type KubemarkContainersSpec struct {
+	// Docker image name for kubemark.
+	// If not set gcr.io/k8s-staging-cluster-api/kubemark-amd64:<kubernetes version> will be used
+	// +optional
+	Image *string `json:"image,omitempty"`
+
+	// Arguments to the kubemark entrypoint.
+	// Please note that --morph=kubelet and --name=<node name> will be always added to the list.
+	// +optional
+	Args []string `json:"args,omitempty" protobuf:"bytes,4,rep,name=args"`
+
+	// Compute Resources required by kubemark.
+	// if not specified cpu and memory requests, defaults of 40m and 10240Ki will be appied.
+	// +optional
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
```

Comment:
this is the resource requirements to run kubemark?

Reply:
Yes, those are resources for the container running the hollowkubelet